pennylane/devices/qubit/sampling.py

```diff
@@ -471,67 +479,60 @@ def sample_state(
     Returns:
         ndarray[int]: Sample values of the shape (shots, num_wires)
     """
-    if prng_key is not None or qml.math.get_interface(state) == "jax":
-        return _sample_state_jax(
-            state, shots, prng_key, is_state_batched=is_state_batched, wires=wires, seed=rng
-        )
-
-    rng = np.random.default_rng(rng)
 
     total_indices = len(state.shape) - is_state_batched
     state_wires = qml.wires.Wires(range(total_indices))
 
     wires_to_sample = wires or state_wires
     num_wires = len(wires_to_sample)
-    basis_states = np.arange(2**num_wires)
 
     flat_state = flatten_state(state, total_indices)
     with qml.queuing.QueuingManager.stop_recording():
         probs = qml.probs(wires=wires_to_sample).process_state(flat_state, state_wires)
+        # Keep same interface (e.g. jax) as in the device
 
-    # when using the torch interface with float32 as default dtype,
-    # probabilities must be renormalized as they may not sum to one
-    # see https://github.com/PennyLaneAI/pennylane/issues/5444
-    norm = qml.math.sum(probs, axis=-1)
-    abs_diff = qml.math.abs(norm - 1.0)
-    cutoff = 1e-07
+    return sample_probs(probs, shots, num_wires, is_state_batched, rng, prng_key)
 
-    if is_state_batched:
-        normalize_condition = False
 
-        for s in abs_diff:
-            if s != 0:
-                normalize_condition = True
-            if s > cutoff:
-                normalize_condition = False
-                break
+def sample_probs(probs, shots, num_wires, is_state_batched, rng, prng_key=None):
+    """
+    Sample from given probabilities, dispatching between JAX and NumPy implementations.
+    """
+    if JAX_AVAILABLE:
+        is_probs_jax = isinstance(probs, jax.Array)
```

Comment:
`jax.numpy.array` does not return a `jax.Array`, but rather a `jaxlib.xla_extension.ArrayImpl`. Instead, we can use `qml.math.get_interface` here:
```suggestion
        is_probs_jax = qml.math.get_interface(probs) == "jax"